### PR TITLE
Streams resource interface

### PR DESCRIPTION
**Pull request by ppaidi**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: TBD
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

Implemented stream interface panel
	- input resource inputs outputs
	- select an input as group by
	- show selected group by property in interface panel

### CODE REVIEW

File: libs/plugins/stream-client/src/lib/core/models/generate-state-from-resource.ts [hunk 1]
```diff
@@ -25,13 +25,19 @@ export function generateStateFromResource(resource: any, schemas: ContributionSc
...
+    output: [],
+  };
   return {
     id,
     name,
     description,
     app,
     triggers,
     handlers,
+    metadata,
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/core/models/generate-state-from-resource.ts]: may be you want to add `metadata:null` to `INITIAL_STREAM_STATE` in `libs/plugins/stream-client/src/lib/core/state/stream.state.ts`

File: libs/plugins/stream-client/src/lib/core/models/generate-state-from-resource.ts [hunk 2]
```diff
@@ -25,13 +25,18 @@ export function generateStateFromResource(resource: any, schemas: ContributionSc
   const stages = (resource && resource.data && resource.data.stages) || [];
   const items = makeStageItems(stages, schemaDefs);
   const graph = makeGraphNodes(stages);
+  const metadata = resource.metadata || {
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/core/models/generate-state-from-resource.ts]: Initializing the metadata with empty data like the "or" case, we might be adding the metadata with 
```
{
   input: [],
   output: []
}
```
when the metadata was supposed to be null / undefined. We might have to add logic to ignore the input / output or the whole metadata when they are empty while exporting the stream resource.

File: libs/plugins/stream-client/src/lib/core/state/clean-dangling-tasks-output-mappings.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { isEmpty, pick, fromPairs } from 'lodash';
+import { ContributionSchema } from '@flogo-web/core';
+import { Dictionary } from '@flogo-web/lib-client/core';
+import { isMapperActivity } from '@flogo-web/plugins/flow-core';
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/core/state/clean-dangling-tasks-output-mappings.ts]: We should not be loading the logic from `flow-core` as you are mixing the plugin logic internally

File: libs/plugins/stream-client/src/lib/params-schema/params-schema.component.html [hunk 4]
```diff
@@ -0,0 +1,151 @@
+<bs-modal [size]="'lg'" #modal cssClass="flow-input-params-modal">
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/params-schema/params-schema.component.html]: can we replace the bs-modal with the modal library? also can we rename the `flow` classes to `streams`?

File: libs/plugins/stream-client/src/lib/params-schema/params-schema.component.less [hunk 5]
```diff
@@ -0,0 +1,143 @@
+@import '~@flogo-web/assets/styles/_variables';
+
+.flow-input-params-field {
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/params-schema/params-schema.component.less]: We can replace them to the BEM names while refactoring these classes

File: libs/plugins/stream-client/src/lib/params-schema/params-schema.component.html (hunk 4 above)
Comment (fcastill) [reply libs/plugins/stream-client/src/lib/params-schema/params-schema.component.html]: I think it's better to make the switch to the modal lib in another PR, this one is already extensive enough

**Comment by ppaidi** [comment]
Point 1 (I am unable to deselect an already selected input) will tracked in #1152

File: libs/plugins/stream-client/src/lib/params-schema/params-schema.component.ts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+      : this.metadata;
+    this.paramsForm = this.resourceInterfaceBuilderService.createForm(
+      metadata.input,
+      metadata.output
+    );
+    this.modal.open();
+  }
+
+  onInputSchemaModalCancel() {
+    this.groupBy = this.metadata ? this.metadata.groupBy : null;
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/params-schema/params-schema.component.ts]: Can we have it together with the input/output  form initialization?

**Pull request merged**